docs/admin/workspaceproxies.md

```diff
@@ -0,0 +1,102 @@
+
+Workspace Proxies are a feature to provide low latency experiences for geo-distributed teams.
+
+Coder's networking does a best effort to make direct connections to a workspace. In situations where this is not possible, such as [dashboard connections](../networking/README.md#dashboard-connections), workspace proxies are able to reduce the amount of distance the network traffic needs to travel.
+
+A workspace proxy is a relay connection a developer can choose to use when connecting with their workspace over ssh, a workspace app, port forwarding, etc.
+
+<!-- TODO: Might want to modify this diagram? -->
+
+![ProxyDiagram](../images/workspaceproxy/proxydiagram.png)
+
+# Deploy a workspace proxy
+
+Each workspace proxy should be a unique instance. At no point should 2 workspace proxy instances share the same authentication token.
+
+Workspace proxies can be used in the browser by navigating to the user `Account -> Workspace Proxy`
+
+## Requirements
+
+- The [Coder CLI](../cli.md) must be installed and authenticated as a user with the Owner role.
+
+## Step 1: Create the proxy
+
+Create the workspace proxy and make sure to save the returned authentication token for said proxy. This is the token the workspace proxy will use to authenticate back to primary coderd.
+
+```bash
+$ coder proxy create --name=newyork --display-name="USA East" --icon="/emojis/2194.png"
+Workspace Proxy "newyork" created successfully. Save this token, it will not be shown again.
+Token: 2fb6500b-bb47-4783-a0db-dedde895b865:05271b4ef9432bac14c02b3c56b5a2d7f05453718a1f85ba7e772c0a096c7175
+```
+
+To verify it was created.
+
+```bash
+$ coder proxy ls
+NAME         URL                    STATUS STATUS
+newyork                             unregistered
+```
+
+## Step 2: Deploy the proxy
+
+Deploying the workspace proxy will also register the proxy with coderd and make the workspace proxy usable. If the proxy deployment is successful, `coder proxy ls` will show an `ok` status code:
+
+```
+$ coder proxy lsPM
+NAME              URL                           STATUS STATUS
+brazil-saopaulo   https://brazil.example.com  ok
+europe-frankfurt  https://europe.example.com  ok
+sydney            https://sydney.example.com  ok
+```
+
+Other Status codes:
+
+- `unregistered` : The workspace proxy was created, and not yet deployed
+- `unreachable` : The workspace proxy was registered, but is not responding. Likely the proxy went offline.
+- `unhealthy` : The workspace proxy is reachable, but has some issue that is preventing the proxy from being used. `coder proxy ls` should show the error message.
+- `ok` : The workspace proxy is healthy and working properly!
+
+### Configuration
+
+<!--
+ I am not sure the best way to present this.
+ Ideally in the future we can auto sync some of the settings with coderd.
+ -->
```

Comment:
How are these autogenerated in comparison with https://coder.com/docs/v2/latest/cli/server?

Reply:
Good question

Reply:
Ok just checked. Because the command is hidden the docs are not generating.

https://github.com/coder/coder/blob/c700a7324e400bd7f9c80475f33e8c1746b2d4f0/scripts/clidocgen/gen.go#L116-L118

https://github.com/coder/coder/blob/c700a7324e400bd7f9c80475f33e8c1746b2d4f0/enterprise/cli/workspaceproxy.go#L18-L18

Should we unhide the command now?

Reply:
Yeah, I'm down to un-hide it. Will it also show that it expects stuff like `CODER_ACCESS_URL` and `CODER_HTTP_ADDRESS`?

Reply:
Actually, let's not un-hide it until we have better errors when someone tries it on a non-enterprise license.

Reply:
Marking as resolved then. Those docs will be generated when we unhide. We can add more usage messages when we do that.